### PR TITLE
fix bug: rename duplicated loss name

### PR DESCRIPTION
**Pull request by ZihengJiang**
While users use a shared layer to get two different outputs but have one same name, the batch_logs/epoch_logs will only left one loss, because it's a `dict`.

```
import numpy as np
from keras.models import *
from keras.layers import *

train_d = np.random.rand(128, 64)
test_d  = np.random.rand(2, 128, 32)

x = Input(shape=(64, ))
a = Dense(32)(x)
b = Dense(32)(x)

shared_dense = Dense(32, name='shared_dense')
out1 = shared_dense(a)
out2 = shared_dense(b)

model = Model(input=x, output=[out1, out2])
model.compile(optimizer='rmsprop', loss='mse')

his = model.fit(train_d, [test_d[0], test_d[1]])
```

The code above will get output like below, but it should have two shared_dense_loss.

image

To solve this problem, I rename the duplicates name by appending it a number. Then it works well:

image

### CODE REVIEW

File: keras/engine/training.py [hunk 1]
```diff
@@ -1040,6 +1040,10 @@ def fit(self, x, y, batch_size=32, nb_epoch=10, verbose=1, callbacks=[],
 
         # prepare display labels
         out_labels = self.metrics_names
+
+        # rename duplicated metrics name
+        out_labels = list(map(lambda x: x[1] + str(out_labels[:x[0]].count(x[1]) + 1) if out_labels.count(x[1]) > 1 else x[1], enumerate(out_labels)))
```
Comment (fchollet) [inline keras/engine/training.py]: I think this line is very long and very difficult to read. It should be rewritten.

**Comment by ZihengJiang** [comment]
@fchollet Hi, I just rewrite the code. What do you think of it?

**Comment by fchollet** [comment]
LGTM
